### PR TITLE
[ECP-8954] Automate the Shopware Marketplace zip file creation as release asset

### PR DESCRIPTION
**Pull request by SushmitaThakur**
Creates a new Github Action Workflow that gets triggers on new release, creates the zip file of the plugin and uploads it into the latest release as an asset.

### CODE REVIEW

File: .github/workflows/upload-release-asset.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+on:
+  release:
+    types: [created, edited]
```
Comment (candemiralp) [inline .github/workflows/upload-release-asset.yml:3]: `edited` might be tricky. If `edited` is triggered this workflow again, any text or description change might result in a new file. 

Do we really need to trigger it when a release is updated? Generally, we don't update tag version. That means files will not change in any case on a release.

File: .github/workflows/upload-release-asset.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          echo "tag_name=$tag_name" >> $GITHUB_OUTPUT
+          echo "Latest Release ID: $release_id"
+          echo "Latest Release Upload URL: $upload_url"
+          echo "Latest Release Tag name: $tag_name"
+
+      - name:  Upload asset
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release upload ${{ steps.get-release-info.outputs.tag_name }} AdyenPaymentShopware6.zip --clobber
```
Comment (candemiralp) [inline .github/workflows/upload-release-asset.yml:37]:
```suggestion
          gh release upload ${{ steps.get-release-info.outputs.tag_name }} AdyenPaymentShopware6.zip --clobber
          
```

File: .github/workflows/upload-release-asset.yml (hunk 1 above)
Comment (SushmitaThakur) [reply .github/workflows/upload-release-asset.yml:3]: Oh, if that is the case, you are right we don't need it on 'edited'. I assumed that we might have scenarios where release might have file changes.

**Review by candemiralp — APPROVED** [review]

**Review by khushboo-singhvi — APPROVED** [review]

File: .github/workflows/upload-release-asset.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+on:
+  release:
+    types: [created]
+  workflow_dispatch:
```
Comment (candemiralp) [inline .github/workflows/upload-release-asset.yml:4]: `workflow_dispatch` event might create ambiguous behaviours if the latest tag is different than the branch on which dispatch is triggered.
```suggestion
```

**Review by candemiralp — APPROVED** [review]

**Pull request merged**